### DISCUSSION TITLE
Update AssociatedOwnerGroup

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#782] [Q&A]
Is it possible to update Web AssociatedOwnerGroup using PnP.Core.SDK. Property AssociatedOwnerGroup in IWeb has getter only and no setter.

Thank you in advance!

**Comment by jansenbe** [comment]
@bajce : currently this is not possible in PnP Core SDK. We did support this PnP Framework but do recall this wasn't always very reliable and its quite an edge case. We don't feel confident that at this point it's a must have feature for PnP Core SDK, but happy to be proven wrong if you've a compelling need and no workaround.

**Reply by bajce** [reply]
@jansenbe: Our use case is that we are programmatically switching a securable object to unique permissions and we need to mimic the SharePoint UI behavior: create new groups and set them as associated groups.

The workaround would be to get the CSOM ClientContext and do the calls there, but that would require us to [include the PnP.Framework ](https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html#using-pnp-framework-when-the-pnp-core-sdk-was-already-configured) in our app, which we would like to avoid if possible.

It would be great if you could add it to the PnP.Core.SDK! 
Thanks in advance for your consideration.

**Reply by jansenbe** [reply]
Can you add an issue to track this need @bajce, don't have bandwidth for it the coming days but maybe someone else jumps on this.

**Reply by bajce** [reply]
@jansenbe https://github.com/pnp/pnpcore/issues/784 I hope I did it right 😄

**Reply by bajce** [reply]
@jansenbe: I tried couple of workarounds and it looks like none of them is working:

Using PnP.Core.SDK:

I found out that associated owner group is not only saved in `Web.AssociatedOwnerGroup` but also in `Web.AllProperties["vti_associateownergroup"]` so I tries to update that value as follow:

 ```
var webAllProperties = context.Web.AllProperties;
var associateOwnerGroup = webAllProperties["vti_associateownergroup"];

webAllProperties["vti_associateownergroup"] = 13;
webAllProperties.Update();
```

I get following error:

```
HttpResponseCode: 200
Message: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
ErrorTypeName: "System.UnauthorizedAccessException
```

I tried to update some other Web properties (to make sure im privileged edit this Web) such as description and that works.

Using PnP.Framework:

```
using (var csomContext = PnPCoreSdk.Instance.GetClientContext(context))
 {
      var csomContextWeb = csomContext.Web;
      csomContext.Load(csomContextWeb,
                   w => w.Id,
                   w => w.Url,
                   w => w.SiteGroups,
                   w => w.AssociatedOwnerGroup,
                   w => w.RoleDefinitions);
                            
      csomContext.ExecuteQuery();
                            
      var ownerGroup = csomContextWeb.SiteGroups.GetById(ownerGroupId);
      
      // csomContextWeb.AssociatedOwnerGroup = new Group(csomContextWeb.Context, ownerGroup.Path); Also tried this
      // csomContextWeb.AssociatedOwnerGroup = null; Also tried this
      csomContextWeb.AssociatedOwnerGroup = ownerGroup;
      
      // ownerGroup.Update(); Also tried this
      
      csomContextWeb.Update();                  
      csomContext.ExecuteQuery();  
 }
```
I get following error:

`Microsoft.SharePoint.Client.ServerUnauthorizedAccessException: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))`

Using PnP.PowerShell: 

```
$web.AssociatedOwnerGroup = $ownerGroup
$web.Update()
```

`Set-PnPGroup -Identity $ownerGroup -SetAssociatedGroup Owners -AddRole "XXX"`

I get following error:
`Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))`

For all tree approaches im using certificate authentication for application.
